### ISSUE TITLE
Megasat SAT>IP Server 2 

### ISSUE THREAD

**Issue by danielkucera** [perexg/satip-axe#137]
Hi @perexg ,
do you think, this device is also supported? I tried to boot from flash but without success.
Can you make an educated guess which port might be the serial console from this picture? Does it look similar to other devices you have worked with?
![img_20180712_155530](https://user-images.githubusercontent.com/1734256/51253819-b9d8d300-199f-11e9-92ab-571ba66a9467.jpg)

http://cdn-reichelt.de/documents/datenblatt/F100/MAN_SAT-IP_SERVER2_EN.pdf

**Comment by perexg** [comment]
I don't think that it's the compatible device with the axe firmware. I think that the serial port is on one of the white 4-pin connector (I would try the one nearby the DC input at first - usually, those connectors have GND/3V on the border pins).

Have you tried to obtain the GPL software from the vendor? They should provide all GPL software they used in the firmware according the licence to the end user (kernel/uboot etc.).

**Comment by danielkucera** [comment]
Hi @perexg , is the storage layout (1 MB NOR / 128 MB NAND) similar to Telestar? I suppose that the SPI NOR chip contains bootloader. Could you provide a dump from your device?

**Comment by danielkucera** [comment]
Ok, I dumped that nor (attached). It contains:
```
$ binwalk megasat.bin 

DECIMAL       HEXADECIMAL     DESCRIPTION
--------------------------------------------------------------------------------
1048576       0x100000        Squashfs filesystem, little endian, version 4.0, compression:gzip, size: 8355 bytes, 4 inodes, blocksize: 131072 bytes, created: 2014-03-21 11:38:37
2031760       0x1F0090        PEM certificate
```
squashfs contains:
```
$ ls -lah /mnt/tst
total 22K
drwxrwxrwx 2  501  501   58 Mar 21  2014 .
drwxr-xr-x 7 root root 4.0K Jul 23 09:58 ..
-rw-r--r-- 1  501  501  253 Mar 21  2014 crc.sh
-rw-r--r-- 1  501  501  143 Mar 21  2014 ldr.sh
-rwxr-xr-x 1  501  501  17K Mar 21  2014 WDReset
```
some interesting strings from WDReset:
```
/opt/STM/STLinux-2.4/devkit/sh4/lib/gcc/sh4-linux/4.5.3/include
GCC: (GNU) 4.5.3 20110525 (STMicroelectronics/Linux Base 4.5.3-93)
/home/macaroni/users/products/stm2.4/build/packages/stm-target-glibc-sh4/BUILD/glibc-2.10.2/csu
GNU C 4.5.3 20110525 (STMicroelectronics/Linux Base 4.5.3-93)
/dev/mem
```

**Comment by danielkucera** [comment]
[megasat.zip](https://github.com/perexg/satip-axe/files/2774462/megasat.zip)

**Comment by danielkucera** [comment]
[bootlog.txt](https://github.com/perexg/satip-axe/files/2774534/bootlog.txt)

**Comment by danielkucera** [comment]
Ok, I managed to compile and boot u-boot, still need to correctly configure NAND flash
```
Board: STx7108-HDK  [32-bit mode]     
U-Boot 1.3.1-0-g3cbc4aae (Jan 25 2019 - 15:22:52) - stm24_0056-hdk7108

DRAM:  128 MiB
ERROR: flash_write_cmd() ignoring write request (info->portwidth=16)
## Unknown FLASH on Bank 1 - Size = 0x00000000 = 0 MiB
NOR:     0 KiB
NAND:  nand_bbt: ECC error while reading bad block table
128 MiB
*** Warning - bad CRC, using default environment

In:    serial
Out:   serial
Err:   serial
HDK7108> 
HDK7108> 
HDK7108> 
HDK7108>
```

**Comment by danielkucera** [comment]
Ok, I'm much further today. I have booted kernel with loaded drivers, working ethernet and running minisatip8:
```
/ # minisatip8 -f &
/ # [01/01 12:01:16.027 main]: getlocalip: Error 'Network is unreachable'' during connect
@@@@@@@@@@@@ fe_open 0
@@@@@@@@@@@@ fe_open 1
@@@@@@@@@@@@ fe_open 2
@@@@@@@@@@@@ fe_open 3
@@@@@@@@@@@@ fe_open 0
@@@@@@@@@@@@ fe_open 1
@@@@@@@@@@@@ fe_open 2
@@@@@@@@@@@@ fe_open 3
[01/01 12:01:16.107 main]: Initializing with 4 devices

/ # [dms-ts] stream stop ok
[dms-ts] stream stop ok
[dms-ts] stream stop ok
[dms-ts] stream stop ok
```
Now I need to find a sat dish with LNB to test if it really works...
@perexg , are you interrested in pulling those changes with instructions into your repo or should I maintain my own fork?

**Comment by danielkucera** [comment]
@perexg , booting stucks on following line for me:
`nc -l 127.0.0.1:1001 -e /bin/true`
https://github.com/perexg/satip-axe/blob/master/fs-add/etc/init.d/rcS#L124

How should this work?

**Comment by perexg** [comment]
https://github.com/perexg/satip-axe/blob/master/fs-add/root/main_init.sh

**Comment by danielkucera** [comment]
Thanks @perexg ,
this fixes my issue:
```
@@ -121,7 +130,7 @@ for i in $(cat /proc/devices); do
   maj=
 done
 /etc/init.d/axe &
-nc -l 127.0.0.1:1001 -e /bin/true
+nc -l -p 1001 -e /bin/true
 rm -f /root/main_init.sh /tmp/axe-done
 logger -p local0.notice "AXE drivers load finish"
```
**What version of nc are you using that it works?** Mine is:
```
/ # nc
BusyBox v1.29.3 (2019-01-28 00:01:15 UTC) multi-call binary.
```

**Comment by danielkucera** [comment]
And the same here:
```
diff --git a/fs-add/etc/init.d/satip b/fs-add/etc/init.d/satip
index 9513572..a8ebc49 100755
--- a/fs-add/etc/init.d/satip
+++ b/fs-add/etc/init.d/satip
@@ -24,7 +24,7 @@ fi
 cd /tmp
 while test ! -f /tmp/nosatip; do
   while ! test -r /tmp/satip-network; do
-    nc -l 127.0.0.1:999 -e /etc/init.d/satip-network
+    nc -l -p 999 -e /etc/init.d/satip-network
   done
   MINISATIP_PROGRAM=minisatip
   MINISATIP5_PROGRAM=minisatip5
```

**Comment by danielkucera** [comment]
my nc accepts following parameters:
```
/ # nc
BusyBox v1.29.3 (2019-01-28 00:01:15 UTC) multi-call binary.

Usage: nc [OPTIONS] HOST PORT  - connect
nc [OPTIONS] -l -p PORT [HOST] [PORT]  - listen

        -e PROG Run PROG after connect (must be last)
        -l      Listen mode, for inbound connects
        -lk     With -e, provides persistent server
        -p PORT Local port
        -s ADDR Local address
        -w SEC  Timeout for connects and final net reads
        -i SEC  Delay interval for lines sent
        -n      Don't do DNS resolution
        -u      UDP mode
        -v      Verbose
        -o FILE Hex dump traffic
        -z      Zero-I/O mode (scanning)
```

**Comment by perexg** [comment]
<pre>
~ # nc
BusyBox v1.26.2 (2018-01-25 08:55:26 CET) multi-call binary.

Usage: nc [-iN] [-wN] [-l] [-p PORT] [-f FILE|IPADDR PORT] [-e PROG]

Open a pipe to IP:PORT or FILE

	-l	Listen mode, for inbound connects
		(use -ll with -e for persistent server)
	-p PORT	Local port
	-w SEC	Connect timeout
	-i SEC	Delay interval for lines sent
	-f FILE	Use file (ala /dev/ttyS0) instead of network
	-e PROG	Run PROG after connect
</pre>

**Comment by paluseb** [comment]
@danielkucera I bought this device but it won't start. The device after switching on is looking for new software on the internet because it is set up but it does not find it and later nothing happens. What can I do? Which connector is rs232 and what should I upload to start the server and get to the settings?

**Comment by danielkucera** [comment]
Not a good device to buy.
![IMG_20200116_192353](https://user-images.githubusercontent.com/1734256/72552023-54f7d780-3896-11ea-8a1d-a4e8343df80a.jpg)

**Comment by danielkucera** [comment]
Connect some usb to serial (TTL) adapter and upload the log here.

**Comment by paluseb** [comment]
I turned on the device, then connected the network cable and then connected the pendrive with the new software. Sorry for my English
[bootlog.txt](https://github.com/perexg/satip-axe/files/4073308/bootlog.txt)

**Comment by paluseb** [comment]
Why "Not a good device to buy."
I uploaded a log on github.



czw., 16 sty 2020 o 19:45 Daniel Kucera <notifications@github.com>
napisał(a):

**Comment by danielkucera** [comment]
Because it's not supported by the manufacturer nor by satip-axe.

these lines right at the begining are suspicious:
```
LOADER.C 4273> Verifying Upgrader Group [1]
Erasing FlashRom
Programming Image
LOADER.C 1494>Preparing to start Application
```

do you have some tool for SPI flash programming? I can send you alternative u-boot I compiled.

**Comment by paluseb** [comment]
I don't have a programmer. But I may be looking for someone who has the right equipment and skills. If possible, send these files and commands that need to be programmed.

**Comment by danielkucera** [comment]
What's your goal? To run the original firmware or my not yet working version of satip-axe?

**Comment by paluseb** [comment]
At the beginning I want to run the original firmware. I don't know that later. Can your software run with a usb stick?

**Comment by danielkucera** [comment]
It can run from USB but not without modifying and disabling the original software.
Try USB upgrade from here 
https://www.megasat.tv/en/support/downloads-de/download-info/satip-server-2-firmware/

**Comment by paluseb** [comment]
I tried it but it doesn't work. However, I noticed an error here:
`webremote/advertise.c 914>Advertise Created Successfully{Tsk[0x1c1c3cc] Ret[0] Sudp[38]}ETH LinkUpStatus...: Linkup
IP Settings Mode...: Manual
DHCP Status........: Success
IP Addr............: 192.168.5.10
NetMask............: 255.255.255.0
Gateway Addr.......: 192.168.5.254
PriDns.............: 192.168.5.254
Sec Dns.............: 0.0.0.0
MAC Addr...........: B8:87:1E:**0**:AF:55.`
[log1.txt](https://github.com/perexg/satip-axe/files/4074298/log1.txt)


After changing the network settings (pc-server lan connection) I uploaded the loader again and now I have such a log.
[log3.txt](https://github.com/perexg/satip-axe/files/4074306/log3.txt)

Unfortunately, megasat is not connecting to the network now. As if it wasn't downloading settings from a router.

**Comment by danielkucera** [comment]
So does everything work ok now?

**Comment by paluseb** [comment]
Yes, probably yes. I haven't connected the antenna yet but I can make all settings on the device

**Comment by burakozhan** [comment]
Has anyone a firmware that I can introduce via USB to this device. 
I do own a second hand device that has a corrupted flash: 
```
ERR>CRC Act [59CC3E17] Exp[DABF2C53] IS[ 2127346 ] Id [ 11 ]
LOADER.C 1628>Flash verification failed
```
I still have working serial console. When connected to a DHCP network it constantlly re-tries to download a valid firmware from some long gone Megasat server. 
<details>
<summary>Log of Connect loop</summary>

```
NDSIGNAL.C 904> Initializing session param
SNETDWNLD.C 2068> Recvr Model : GMI_7108/MODELF

ONDSIGNAL.C 927> Connect to server [59.120.115.74]

NETUPGRD.C 158> {CONNCT.Rqst} Prtcl[1] Srv.IP[59.120.115.74] SWMdl[GMI_7108/MODELF]
ONDSIGNAL.C 947> Wait for netstat nofification

HTTPCLIENT.C 4098> RqstMthd [5] URL[https://59.120.115.74/GMI_7108/MODELF/]
ENETSOCK.C 3868> {ConnFaild} Connect with server failure GsockOpt Errno[111] 
HTTPCLIENT.C 787> Failed to Connect with server !!!!!! 

HTTPCLIENT.C 882> Failed to connect to http server
NETUPGRD.C 433> {Conn.Rqst [Ack]} Res.Code[1]
HTTPCLIENT.C 4160> Reqst Fail Err.code [202] Srvr Res.Code[0]
NETUPGRD.C 433> {Conn.Rqst [Ack]} Res.Code[4821600]
HTTPCLIENT.C 4047> Http Ctrl thread waiting for msg
ONDSIGNAL.C 964> Network download incoming status : 259

ONDSIGNAL.C 972> Failed to connect to server [59.120.115.74]
HTTPCLIENT.C 4504> Http handle closed
ONDSIGNAL.C 827> S/W Not Avail in [59.120.115.74]
ONDSIGNAL.C 342> OND TRacking Method [5]
ONDSIGNAL.C 627> {TRACK CN} IF[2] Mode[1] SerIP [1249081403] 
SATDISCOV.C 1545> Timeout select ERR
```


</details>

I would like to make this box happy again. (The link mentinoned on Jan 16, 2020 is long gone, there is no backup on archive.org)

**Comment by 9000h** [comment]
https://github.com/perexg/satip-axe/issues/137#issuecomment-455682962

**Comment by burakozhan** [comment]
Thanks @9000h That file seems to be a binary nor dump. I am looking for the full zip that was once available at the megasat website. 
https://www.megasat.tv/en/support/downloads-de/download-info/satip-server-2-firmware/

The megasat.bin up here I can not re-install via USB. 
I only have access via UART. There is an intermittent HTTP server showing up at http://(DHCP-Box-IP)/upgrader/ which only shows up briefly (like 2-3 seconds) I have buildt a script to upload the bin there, but in both cases(usb and http) there is a checsum mismatch and it will refuse the bin file. 
I really need that full updater zip from megasat.

I will try to get a SOIC Clip to burn the image directly, but I doubt it will solve the problems I have

**Comment by danielkucera** [comment]
https://upload.nolog.cz/download/2bc520c77c4478ef/#0a8WwReYING4gvb9zcNW4Q

**Comment by burakozhan** [comment]
Thank you for the upload! Saved my bricked device. 
Probably previous owner did turn it off during a firmware upgrade. The megasat server is gone therefore the self-rescue system also failed.

For anyone reading in the future:
- There are 2 Files in the zip which the included PDF instructs to put both on a USB stick. DON'T do that.
- With 2 files it gets stuck "Waiting for USB file selection from MMI" But there is no way for a user to input anything.
- Only copy the "0107_WLDR_WAPPL_WMES.usb" into a "UPDATE" folder on the usb. Single file and it will auto-choose it.

Also sorry for getting back into a 6 years old thread. And sorry it does not really help the satip-axe project itself.

**Comment by 9000h** [comment]
Schweiger MS41ip should be the same hardware
http://manual.schwaiger.de/software/ms41ip/software/
